edgelet/edgelet-docker/src/notary.rs

```diff
@@ -0,0 +1,190 @@
+        ))
+    })?;
+
+    // Add https to hostname
+    let input_url_https = format!("https://{}", registry_server_hostname);
+    debug!("URL with https is {}", input_url_https);
+
+    // Create Notary Config.json contents
+    let config_contents = json!({
+      "trust_dir": trust_dir,
+      "remote_server": {
+        "url" : input_url_https,
+      },
+      "trust_pinning": {
+        "ca": {
+          "": path,
+        },
+        "disable_tofu": "true"
+      }
+    });
+    debug!("Config JSON contents {}", config_contents);
+
+    // Generate Notary Config.json path
+    let mut config_file_path = trust_dir;
+    config_file_path.push(r"config.json");
+    debug!("Config file path {}", config_file_path.display());
+
+    // Create Notary config file
+    let file = File::create(&config_file_path).with_context(|_| {
+        ErrorKind::InitializeNotary(format!(
+            "could not create notary config file in {}",
+            config_file_path.display()
+        ))
+    })?;
+    serde_json::to_writer(file, &config_contents).with_context(|_| {
+        ErrorKind::InitializeNotary(format!(
+            "could not create notary config file in {}",
+            config_file_path.display()
+        ))
+    })?;
+    debug!("Config JSON file created successfully");
+    Ok(config_file_path)
+}
+
+pub fn notary_lookup(
+    notary_auth: Option<&str>,
+    image_gun: &str,
+    tag: &str,
+    config_path: &Path,
+    lock: tokio::sync::lock::LockGuard<BTreeMap<String, String>>,
+) -> impl Future<
+    Item = (
+        String,
+        tokio::sync::lock::LockGuard<BTreeMap<String, String>>,
+    ),
+    Error = Error,
+> {
+    let mut notary_cmd = Command::new("notary");
+    if let Some(notary_auth) = notary_auth {
+        notary_cmd.env("NOTARY_AUTH", notary_auth);
+    }
+    notary_cmd
+        .arg("lookup")
+        .args(&[image_gun, tag])
+        .arg("-c")
+        .arg(config_path)
+        .output_async()
+        .then(|notary_output| {
+            let notary_output = notary_output.with_context(|_| {
+                ErrorKind::LaunchNotary("could not spawn notary process".to_owned())
+            })?;
+            let notary_output_string =
+                String::from_utf8(notary_output.stdout).with_context(|_| {
+                    ErrorKind::LaunchNotary("could not retrieve notary output as string".to_owned())
+                })?;
+            debug!("Notary output string is {}", notary_output_string);
+            let split_array: Vec<&str> = notary_output_string.split_whitespace().collect();
+            if split_array.len() < 2 {
+                return Err(ErrorKind::LaunchNotary(
+                    "notary digest split array is empty".to_owned(),
+                )
+                .into());
+            }
+
+            // Notary Server output on lookup is of the format [tag, digest, bytes]
+            Ok((split_array[1].to_string(), lock))
+        })
+}
+
+#[cfg(test)]
```

Comment:
Why are these tests Linux-only?

Reply:
Targeting only Linux for now

Reply:
The code isn't Linux-only.

Reply:
Yea since we are supporting only Linux, I made it all for Linux so its easier to focus on dotnet checkins for linux platforms alone.

Reply:
https://github.com/Azure/iotedge/blob/627acd2e7efa6b137c2586ca48631f5308f8ae41/edgelet/edgelet-docker/src/notary.rs#L12-L14

The code isn't Linux-only. Why are the tests Linux-only?